hedera-node/hedera-networkadmin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeHandler.java

```diff
@@ -54,77 +61,136 @@ public FreezeHandler() {
         // Dagger2
     }
 
+    /**
+     * This method is called during the pre-handle workflow for Freeze transactions.
+     *
+     * @param context the {@link PreHandleContext} which collects all information that will be
+     *     passed to {@code handle()}
+     * @see <a href="https://hashgraph.github.io/hedera-protobufs/#freeze.proto">Protobuf freeze documentation</a>
+     */
     @Override
-    @SuppressWarnings("java:S1874") // disable the warnings for use of deprecated code
-    // it is necessary to check getStartHour, getStartMin, getEndHour, getEndMin, all of which are deprecated
-    // because if any are present then we set a status of INVALID_FREEZE_TRANSACTION_BODY
     public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
+        pureChecks(context.body());
+
+        // from proto specs, it looks like update file not required for FREEZE_UPGRADE and TELEMETRY_UPGRADE
+        // but specs aren't very clear
+        // current code in FreezeTransitionLogic checks for the file in specialFiles
+        // so we will do the same
+        final FreezeTransactionBody freezeTxn = context.body().freezeOrThrow();
+        final FreezeType freezeType = freezeTxn.freezeType();
+        if (Arrays.asList(FREEZE_ONLY, FREEZE_UPGRADE, TELEMETRY_UPGRADE).contains(freezeType)) {
+            final Timestamp txValidStart = context.body().transactionIDOrThrow().transactionValidStartOrThrow();
+            verifyFreezeStartTimeIsInFuture(freezeTxn, txValidStart);
+        }
+        if (Arrays.asList(FREEZE_UPGRADE, TELEMETRY_UPGRADE, PREPARE_UPGRADE).contains(freezeType)) {
+            final ReadableSpecialFileStore specialFileStore = context.createStore(ReadableSpecialFileStore.class);
+            verifyUpdateFileAndHash(freezeTxn, specialFileStore);
+        }
 
-        FreezeTransactionBody freezeTxn = context.body().freeze();
+        // no need to add any keys to the context because this transaction does not require any signatures
+        // it must be submitted by an account with superuser privileges, that is checked during ingest
+    }
+
+    /**
+     * Performs checks independent of state or context
+     */
+    @SuppressWarnings("java:S1874") // disables the warnings for use of deprecated code
```

Comment:
This deprecation warning doesn't block progress, and we generally want to have these warnings displayed so we can more easily tell what needs changing before we can remove the deprecated code path.
Not a requirement, but would be nice to "unsuppress" this "soon" (though perhaps not in this PR).

Reply:
The deprecated methods are in the protobufs, and hence need to stay there forever for backwards compatibility. The only thing this code does is to check that the submitted transaction has not set these deprecated fields. I expect this code will stay here forever (or at least as close to forever as ever exists in software).

Reply:
hmmm.  That's less than ideal, but I see the difficulty.
Probably something to take note of and seek a better option long term.